martin/src/srv/tiles_info.rs

```diff
@@ -26,18 +27,23 @@ async fn get_source_info(
     req: HttpRequest,
     path: Path<SourceIDsRequest>,
     sources: Data<TileSources>,
+    srv_config: Data<SrvConfig>,
 ) -> ActixResult<HttpResponse> {
     let sources = sources.get_sources(&path.source_ids, None)?.0;
 
-    // Get `X-REWRITE-URL` header value, and extract its `path` component.
-    // If the header is not present or cannot be parsed as a URL, return the request path.
-    let tiles_path = req
-        .headers()
-        .get("x-rewrite-url")
-        .and_then(|v| v.to_str().ok())
-        .and_then(|v| v.parse::<Uri>().ok())
-        .map_or_else(|| req.path().to_owned(), |v| v.path().to_owned());
-
+    // if base_path is set, use it as the tiles path
+    // or try to use x-rewrite-url header value as the tiles path
+    let tiles_path = if srv_config.base_path.is_some() {
```

Comment:
rule of thumb:  if you have `.unwrap()`, you are probably doing something bad (unless its a unit test)...  In this case - use `if let Some(base_path) = srv_config.base_path { ... } else { ... }`